source/extensions/load_balancing_policies/subset/subset_lb.cc

```diff
@@ -828,12 +828,12 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const HostHashSet& matching_host
   auto excluded_hosts_per_locality =
       original_host_set_.excludedHostsPerLocality().filter({cached_predicate})[0];
 
-  HostSetImpl::updateHosts(HostSetImpl::updateHostsParams(
-                               hosts, hosts_per_locality, healthy_hosts, healthy_hosts_per_locality,
-                               degraded_hosts, degraded_hosts_per_locality, excluded_hosts,
-                               excluded_hosts_per_locality),
-                           determineLocalityWeights(*hosts_per_locality), hosts_added,
-                           hosts_removed, absl::nullopt, absl::nullopt);
+  HostSetImpl::updateHosts(
+      HostSetImpl::updateHostsParams(
+          hosts, hosts_per_locality, healthy_hosts, healthy_hosts_per_locality, degraded_hosts,
+          degraded_hosts_per_locality, excluded_hosts, excluded_hosts_per_locality),
+      determineLocalityWeights(*hosts_per_locality), hosts_added, hosts_removed, absl::nullopt,
+      original_host_set_.overprovisioningFactor());
```

Comment:
**Old behavior**
When a subset load balancer is created, a priority set with the default overprovisioning factor is passed into the constructor here https://github.com/envoyproxy/envoy/blob/54fd807063590190ac31850a9be0d64fbb33851e/source/common/upstream/cluster_manager_impl.cc#L1560-L1567. The subset load balancer then uses that overprovisioning factor value when creating a priority subset here https://github.com/envoyproxy/envoy/blob/54fd807063590190ac31850a9be0d64fbb33851e/source/extensions/load_balancing_policies/subset/subset_lb.cc#L876. The overprovisioning factor used by the subset load balancer will not be updated when the original host set's overprovisioning factor is updated, because the call to `HostSetImpl::updateHosts` above^ uses `absl::nullopt` for the overprovisioning factor parameter.

**New behavior**
The overprovisioning factor used by the subset load balancer _will_ be updated when the original host set's overprovisioning factor is updated, because the original host set's overprovisioning factor is passed in when calling `HostSetImpl::updateHosts` above^.